File: howSumTabulation.js
const howSumTabulation = (target, numbers) => {
  const table =  Array(target + 1).fill(null)
  table[0] = []

  for (let i = 0; i <= target; i++){
    if (table[i] !==  null ){
     for (let num of  numbers){
      table[i + num] = [...table[i]]
     }
    }
  }
  return table[target]
}
console.log(howSumTabulation(5,[4,5,3]))

File: howSumTabulation.rb
def how_sum_tabulation(target, numbers)
  table = Array.new(target + 1).fill(nil)
  table[0] = []
  i = 0
 until i < target
    i+=1
    if table[i] != nil
      numbers.each do |num|
        table[i + num] = [*table[i], num]
        end
    end
  end
  return table[target]
end

how_sum_tabulation(9, [4,3])
